src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -133,10 +134,21 @@ class WorkspaceService(protected val argUserToken: WithAccessToken, val rawlsDAO
     }
   }
 
-  def updateWorkspaceACL(workspaceNamespace: String, workspaceName: String, aclUpdates: Seq[WorkspaceACLUpdate], originEmail: String, inviteUsersNotFound: Boolean) = {
+  def updateWorkspaceACL(workspaceNamespace: String, workspaceName: String, aclUpdates: Seq[WorkspaceACLUpdate], originEmail: String, originId: String, inviteUsersNotFound: Boolean): Future[RequestComplete[(WorkspaceACLUpdateResponseList, Seq[Share])]] = {
+    def logShares(aclUpdateList: WorkspaceACLUpdateResponseList) = {
+      // this will log a share every time a workspace is shared with a user
+      // it will also log a share every time a workspace permission is changed
+      // i.e. READER to WRITER, etc
+      val sharees = aclUpdateList.usersUpdated.filterNot(_.accessLevel == WorkspaceAccessLevels.NoAccess).map(_.email)
+      shareLogDAO.logShares(originId, sharees, ShareType.WORKSPACE)
+    }
+
     val aclUpdate = rawlsDAO.patchWorkspaceACL(workspaceNamespace, workspaceName, aclUpdates, inviteUsersNotFound)
+
     aclUpdate map { actualUpdates =>
-      RequestComplete(actualUpdates)
+      val shares = logShares(actualUpdates)
+      // todo swagger 2.0 doesn't support arrays of mixed types. will need to revisit
+      RequestComplete((actualUpdates, shares))
```

Comment:
why does this need to change the response payload at all? Is there a reason to return `shares` in the response?

Reply:
It was strictly so that I could make a test assertion in `WorkspaceApiServiceSpec`. Is that assertion worth having in your opinion?

Reply:
I do like being able to test it. BUT - since changing the response payload affects end users, and this is a breaking API change, I don't think it's worth it. We should refactor code to find a way to test `logShares` without affecting the end user.

Reply:
technically `ElasticSearchShareLogDAO.logShares` is unit tested, and this is really just testing that the MockShareLogDAO endpoint is called. Do we need more test coverage than that?